lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -138,6 +138,29 @@ def set_request_tags!(trace, request_span, env, status, headers, response, origi
             request_span.set_tag(Tracing::Metadata::Ext::TAG_OPERATION, Ext::TAG_OPERATION_REQUEST)
             request_span.set_tag(Tracing::Metadata::Ext::TAG_KIND, Tracing::Metadata::Ext::SpanKind::TAG_SERVER)
 
+            if status != 404 && (last_route = trace.get_tag(Tracing::Metadata::Ext::HTTP::TAG_ROUTE))
+              last_script_name = trace.get_tag(Tracing::Metadata::Ext::HTTP::TAG_ROUTE_PATH).to_s
```

Comment:
If you are coercing `nil` to `''` I would perform that in the comparison below (on line 150), i.e. `last_script_name || '' == ''` which I think conveys the intent clearer. Or even default that here instead of using `to_s`.

Reply:
fixed [3207b16](https://github.com/DataDog/dd-trace-rb/pull/3849/commits/3207b16ea39d9a5b7af0d7a1ae392058d2c39656)

Reply:
Looking at this again the .to_s should already convert to a non-nil value so the defaulting shouldn't be needed?